function_calling/notebooks/function_calling_guide.ipynb

```diff
@@ -0,0 +1,1216 @@
+{
```

Comment:
Many functions in this notebook are already defined in `function_calling.py` and `tools.py`.
I would remove the definitions from this notebook and just import all those functions from the source files.

Reply:
It would be more tidy and also less error prone. Then you just need to update the `src` files and not worry about the code repetition here.

Reply:
I also see your point in defining all the example tools in this notebook. We could discuss whether it would be useful to have a source file with some example tools that could be used very often by many people, a sort of registry with the most frequent tools. (Maybe to be added in the future?)